pkgs/tools/misc/hdf5/default.nix

```diff
@@ -15,34 +17,45 @@ assert !cpp || mpi == null;
 with { inherit (stdenv.lib) optional optionals; };
 
 stdenv.mkDerivation rec {
-  version = "1.8.16";
+  version = "1.10.0";
   name = "hdf5-${version}";
   src = fetchurl {
-    url = "http://www.hdfgroup.org/ftp/HDF5/releases/${name}/src/${name}.tar.bz2";
-    sha256 = "1ilq8pn9lxbf2wj2rdzwqabxismznjj1d23iw6g78w0bl5dsxahk";
- };
+    url = "http://www.hdfgroup.org/ftp/HDF5/releases/hdf5-1.10/${name}/src/${name}.tar.bz2";
+    sha256 = "0a1xfj73ipykiq01n9y1d6f0k2wa972fx8xxndm0c5vkgkf71zri";
+  };
+
+  patches = [ ./hl-H5LDpublic.patch ];
 
   passthru = {
     mpiSupport = (mpi != null);
     inherit mpi;
   };
 
-  buildInputs = []
+  buildInputs = [ cmake ]
     ++ optional (gfortran != null) gfortran
     ++ optional (zlib != null) zlib
     ++ optional (szip != null) szip;
 
   propagatedBuildInputs = []
     ++ optional (mpi != null) mpi;
 
-  configureFlags = []
-    ++ optional cpp "--enable-cxx"
-    ++ optional (gfortran != null) "--enable-fortran"
-    ++ optional (szip != null) "--with-szlib=${szip}"
-    ++ optionals (mpi != null) ["--enable-parallel" "CC=${mpi}/bin/mpicc"]
-    ++ optional enableShared "--enable-shared";
-
-  patches = [./bin-mv.patch];
+  cmakeFlags = [
+    "-DCMAKE_BUILD_TYPE=Release"
+    "-DCMAKE_C_FLAGS=-fPIC"
+    "-DHDF5_BUILD_HL_LIB=ON"
+    "-DHDF5_DISABLE_COMPILER_WARNINGS=ON"
+    "-DHDF5_BUILD_TOOLS=ON"
+    "-DCMAKE_EXE_LINKER_FLAGS=''"
+    "-DHDF5_INSTALL_DATA_DIR=share/hdf5"
+    "-DHDF5_INSTALL_CMAKE_DIR=share/cmake/hdf5"
+  ]
+    ++ stdenv.lib.optional (mpi != null)  "-DHDF5_ENABLE_PARALLEL=ON -DHDF5_BUILD_CPP_LIB:BOOL=OFF"
```

Comment:
Could the `HDF5_BUILD_CPP_LIB` depend on the `cpp` parameter (see line 5)?

With probably something like `assert mpi -> !cpp`